vnext/JSI/Universal/ChakraRtRuntime.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifndef CHAKRACORE
+
+#include "ChakraRuntime.h"
+#include "ChakraRuntimeArgs.h"
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::JSI {
+
+class ChakraRtRuntime : public ChakraRuntime {
+ public:
+  using ChakraRuntime::ChakraRuntime;
+  ~ChakraRtRuntime() noexcept = default;
+
+#pragma region Functions_inherited_from_Runtime
+
+  facebook::jsi::Value evaluatePreparedJavaScript(
+      const std::shared_ptr<const facebook::jsi::PreparedJavaScript> &preparedJs) override;
+
+  std::string description() override;
+
+ private:
+  // Only ChakraCore supports weak reference semantics, so ChakraRtRuntime
+  // WeakObjects are in fact strong references.
```

Comment:
>WeakObjects are in fact strong references [](start = 5, length = 41)

This worries me a lot, since it seems like we can silently keep objects alive too long and create leaks or hard to debug issues. Do we know if this is used? If not could we maybe terminate() or provide some behavior to make it clear something bad may happen?

Reply:
This looks like existing behavior. Do we have a bug tracking this?

---
In reply to: [345444948](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345444948) [](ancestors = 345444948)

Reply:
We do not have a but tracking this at the moment. I'll create a issue.

---
In reply to: [345450009](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345450009) [](ancestors = 345450009,345444948)